sdcm/cluster.py

```diff
@@ -1135,27 +939,53 @@ def start_task_threads(self):
             if Setup.BACKTRACE_DECODING:
                 self.start_decode_on_monitor_node_thread()
 
+    def get_backtraces(self):
+        if not self._coredump_thread:
+            return
+        self._coredump_thread.get_coredumps(self._coredump_thread.get_core_pids())
+
+    @property
+    def n_coredumps(self):
+        if not self._coredump_thread:
+            return 0
+        return self._coredump_thread.n_coredumps
+
     @log_run_info
-    def stop_task_threads(self, timeout=10):
+    def stop_task_threads(self):
         if self.termination_event.isSet():
             return
         self.log.info('Set termination_event')
         self.termination_event.set()
+        if self._coredump_thread:
+            self._coredump_thread.stop()
+        if self._alert_manager:
+            self._alert_manager.stop()
+
+    @log_run_info
+    def wait_till_tasks_threads_are_stopped(self, timeout: float = 120):
+        await_bucket = []
         if self._spot_monitoring_thread:
-            self._spot_monitoring_thread.join(timeout)
-        if self._backtrace_thread:
-            self._backtrace_thread.join(timeout)
+            await_bucket.append(self._spot_monitoring_thread)
         if self._db_log_reader_thread:
-            self._db_log_reader_thread.join(timeout)
+            await_bucket.append(self._db_log_reader_thread)
         if self._alert_manager:
-            self._alert_manager.stop(timeout)
+            await_bucket.append(self._spot_monitoring_thread)
+        if self._coredump_thread:
+            await_bucket.append(self._coredump_thread)
+        if self._decoding_backtraces_thread:
+            await_bucket.append(self._decoding_backtraces_thread)
+        end_time = time.perf_counter() + timeout
+        while await_bucket and end_time > time.perf_counter():
+            for thread in await_bucket.copy():
+                if not thread.is_alive():
+                    await_bucket.remove(thread)
+            time.sleep(1)
+
```

Comment:
remove `_coredump_thread` from `await_bucket` and wait for upload to finsih

Reply:
done